src/components/Presentation/DataTable.vue

```diff
@@ -26,6 +26,23 @@ export default {
       required: true,
     },
   },
+  data() {
+    return { lastCheckBoxIdx: null };
+  },
+  methods: {
+    handleRowSelect({ shiftKey }, props) {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
In a previous incarnation of this branch I had advocated for disabling this rule globally via:

```json
no-param-reassign: ["error", { "props": false }]
```

How do people feel about that? @subdavis @jbeezley @matthewma7

Reply:
I personally like that rule and think it should be followed whenever possible.  We only broke it here because vuetify implements the `toggle()` behavior through a setter function on `props.selected`.

Reply:
I think the rule is *weird*, because it has two (seemingly) unrelated parts that are lumped into the same top-level rule.

1. No reassignment of params.
2. No *modification* of params.

The second case is there to promote more functional-style programming by disallowing side-effects on input params. The first one has nothing to do with side effects, and is in a completely different category of potential problems. I think the first one is much more in the realm of *linting* and the second is much more about *programming patterns*.

I feel comfortable programming with either of those things disabled, but if others think it might actually catch us doing unintended or bad things, we can keep the default and I can sprinkle in line-level disables.

Reply:
I found this one a little bit weird because it prevented me form doing `Function.prototype={}`

Reply:
Alright, Zach and Matthew have made a good case.  Let's drop this rule.